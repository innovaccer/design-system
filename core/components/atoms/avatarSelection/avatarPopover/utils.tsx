import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
