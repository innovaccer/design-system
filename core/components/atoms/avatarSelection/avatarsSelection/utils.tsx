import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
