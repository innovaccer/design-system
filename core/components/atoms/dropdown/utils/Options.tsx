export let dropdownOptions: any[] = [];

for (let i = 1; i <= 50; i++) {
  dropdownOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
  });
}

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = dropdownOptions.slice(0, 10);

export const storySections = { 0: 'subheading', 5: 'subheading' };

export const subInfoItems = [
  {
    label: 'Option1',
    value: 'Option1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option2',
    value: 'Option2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option3',
    value: 'Option3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option1',
    value: 'Option1',
    icon: 'events'
  },
  {
    label: 'Option2',
    value: 'Option2',
    icon: 'events',
  },
  {
    label: 'Option3',
    value: 'Option3',
    icon: 'events',
  },
];
