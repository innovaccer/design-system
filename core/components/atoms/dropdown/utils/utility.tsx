const getSearchedOptions = (options: any, searchTerm: string) => {
  const result = options.filter((option: any) => option.label.toLowerCase().includes(searchTerm.toLowerCase()));
  return result;
};

export const getOptions = (offset: number, limit: number, searchTerm: string, options: any) => {
  const searchedOptions = searchTerm ? getSearchedOptions(options, searchTerm) : options;
  return new Promise(resolve => {
    resolve({
      offset,
      slicedOptions: searchedOptions.slice(offset, offset + limit),
      options: searchedOptions,
    });
  });
};

export const getValuesFromSelectedObj = (selectedArray: any[] = []) => {
  const selectedValues: any[] = [];

  selectedArray.forEach((selectedObj: any) => {
    const { value } = selectedObj;
    selectedValues.push(value);
  });

  return selectedValues;
};

export const getIndexesFromSelectedObj = (selectedArray: any[] = []) => {
  return (Array.from(Array(selectedArray.length).keys()));
};

export const getValuesFromInd = (selectedArray: number[], options: any[]) => {
  const result: any[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].value);
  });

  return result;
};

export const getLabelsFromInd = (selectedArray: number[], options: any[]) => {
  const result: string[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].label);
  });

  return result;
};
