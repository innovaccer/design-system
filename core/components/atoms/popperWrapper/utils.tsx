// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
