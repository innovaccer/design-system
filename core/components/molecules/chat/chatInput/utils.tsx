import { MessageType } from './ChatInput';

export const positionMentionPopup = (
  textareaRef: React.RefObject<HTMLDivElement>,
  setMentionPosition: React.Dispatch<React.SetStateAction<{ top: number; left: number }>>
) => {
  if (!textareaRef.current) return;

  const selection = window.getSelection();
  if (!selection || !selection.rangeCount) return;

  const range = selection.getRangeAt(0);
  const rect = range.getBoundingClientRect();
  const editorRect = textareaRef.current.getBoundingClientRect();

  // Calculate position based on the cursor position
  let top = rect.bottom + window.scrollY;
  let left = rect.left + window.scrollX;

  // Adjust position if it would go off-screen
  if (top + 150 > window.innerHeight) {
    top = rect.top + window.scrollY - 150;
  }
  if (left + 200 > window.innerWidth) {
    left = window.innerWidth - 200;
  }

  // Adjust position relative to the editor
  top -= editorRect.top + window.scrollY - 12;
  left -= editorRect.left + window.scrollX - 8;

  setMentionPosition({ top, left });
};

export const extractMessageData = (textareaRef: React.RefObject<HTMLDivElement>): MessageType[] => {
  const elements = textareaRef.current?.childNodes;

  return Array.from(elements || []).map((item: any) => {
    if (item?.dataset?.type === 'mention') {
      const mentionValue = JSON.parse(item.dataset.content);
      return { type: 'mention', content: mentionValue?.data, nodeName: item.nodeName };
    }

    // handles new line by pressing enter
    else if (item.innerText === '\n' && item.nodeName === 'DIV') {
      return { type: 'newline', content: '\n', nodeName: item.nodeName };
    }

    // handles line break by pressing shift + enter
    else if (item.nodeName === 'BR') {
      return { type: 'linebreak', content: '<br>', nodeName: item.nodeName };
    }

    return { type: 'text', content: item.innerText || item.textContent, nodeName: item.nodeName };
  });
};
