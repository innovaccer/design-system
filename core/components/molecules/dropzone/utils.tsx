const isIe = (userAgent: string) => {
  return userAgent.indexOf('MSIE') !== -1 || userAgent.indexOf('Trident/') !== -1;
};

const isEdge = (userAgent: string) => {
  return userAgent.indexOf('Edge/') !== -1;
};

export const accepts = (file: File, acceptedFiles?: string | string[]) => {
  if (file && acceptedFiles) {
    const acceptedFilesArray = Array.isArray(acceptedFiles) ? acceptedFiles : acceptedFiles.split(',');
    const fileName = file.name || '';
    const mimeType = (file.type || '').toLowerCase();
    const baseMimeType = mimeType.replace(/\/.*$/, '');

    if (!mimeType) {
      return true;
    }

    return acceptedFilesArray.some((type) => {
      const validType = type.trim().toLowerCase();
      if (validType.charAt(0) === '.') {
        return fileName.toLowerCase().endsWith(validType);
      }
      if (validType.endsWith('/*')) {
        return baseMimeType === validType.replace(/\/.*$/, '');
      }
      return mimeType === validType;
    });
  }
  return true;
};

export const isPropagationStopped = (event: any) => {
  if (typeof event.isPropagationStopped === 'function') {
    return event.isPropagationStopped();
  }
  if (typeof event.cancelBubble !== 'undefined') {
    return event.cancelBubble;
  }
  return false;
};

export const isEvtWithFiles = (event: any) => {
  if (!event.dataTransfer) {
    return !!event.target && !!event.target.files;
  }

  return Array.prototype.some.call(
    event.dataTransfer.types,
    (type) => type === 'Files' || type === 'application/x-moz-file'
  );
};

export const isKindFile = (item: null | Record<string, unknown>) => {
  return typeof item === 'object' && item !== null && item.kind === 'file';
};

export const onDocumentDragOver = (event: DragEvent) => {
  event.preventDefault();
};

export const isIeOrEdge = (userAgent = window.navigator.userAgent) => {
  return isIe(userAgent) || isEdge(userAgent);
};

type functype = (event: Event, ...args: unknown[]) => void;

export const composeEventHandlers = (...fns: functype[]) => {
  return (event: Event, ...args: unknown[]) =>
    fns.some((fn) => {
      if (!isPropagationStopped(event) && fn) {
        fn(event, ...args);
      }
      return isPropagationStopped(event);
    });
};

export const reducer = (state: any, action: any) => {
  switch (action.type) {
    case 'focus':
      return {
        ...state,
        isFocused: true,
      };
    case 'blur':
      return {
        ...state,
        isFocused: false,
      };
    case 'openDialog':
      return {
        ...state,
        isFileDialogActive: true,
      };
    case 'closeDialog':
      return {
        ...state,
        isFileDialogActive: false,
      };
    case 'setDraggedFiles':
      /* eslint no-case-declarations: 0 */
      const { isDragActive, draggedFiles } = action;
      return {
        ...state,
        draggedFiles,
        isDragActive,
      };
    case 'setFiles':
      return {
        ...state,
        acceptedFiles: action.acceptedFiles,
        fileRejections: action.fileRejections,
      };
    case 'reset':
      return {
        ...state,
        isFileDialogActive: false,
        isDragActive: false,
        draggedFiles: [],
        acceptedFiles: [],
        fileRejections: [],
      };
    default:
      return state;
  }
};
