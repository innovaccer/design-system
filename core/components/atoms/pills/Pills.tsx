import * as React from 'react';
import classNames from 'classnames';
import { BaseProps, extractBaseProps } from '@/utils/types';
import { AccentAppearance } from '@/common.type';
import styles from '@css/components/pills.module.css';
import badgeStyles from '@css/components/badge.module.css';

export interface PillsProps extends BaseProps {
  /**
   * Color of the `Pills`
   */
  appearance: AccentAppearance;
  /**
   * Makes `Pills` appearance subtle
   */
  subtle?: boolean;
  /**
   * Text to be added inside `Pills`
   */
  children: React.ReactText;
}

export const Pills = (props: PillsProps) => {
  const { appearance, children, subtle, className } = props;

  const baseProps = extractBaseProps(props);

  const classes = classNames(
    {
      [styles.Pills]: true,
      [badgeStyles[`Badge--${appearance}`]]: appearance && !subtle,
      [badgeStyles[`Badge--subtle-${appearance}`]]: subtle,
    },
    className
  );

  return (
    <span data-test="DesignSystem-Pills" {...baseProps} className={classes}>
      {children}
    </span>
  );
};

Pills.displayName = 'Pills';
Pills.defaultProps = {
  appearance: 'secondary',
};

export default Pills;
