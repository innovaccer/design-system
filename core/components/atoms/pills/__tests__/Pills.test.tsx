import * as React from 'react';
import { render } from '@testing-library/react';
import Pills, { PillsProps as Props, Appearance } from '../Pills';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const appearances: Appearance[] = [
  'primary',
  'alert',
  'warning',
  'success',
  'accent1',
  'accent2',
  'accent3',
  'accent4',
];
const BooleanValue = [true, false];

const mapper = {
  appearance: valueHelper(appearances, { required: true, iterate: true }),
  subtle: valueHelper(BooleanValue, { required: true, iterate: true }),
};

describe('Pills component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Pills {...attr}>Pills</Pills>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Pills component', () => {
  it('renders children', () => {
    const { getByTestId } = render(
      <Pills subtle={true} appearance="secondary">
        {'Pills'}
      </Pills>
    );
    expect(getByTestId('DesignSystem-Pills').textContent).toMatch('Pills');
  });

  describe('Pill component tagName', () => {
    it('renders tagName', () => {
      const { getByTestId } = render(
        <Pills subtle={true} appearance="secondary">
          {'Design System'}
        </Pills>
      );
      expect(getByTestId('DesignSystem-Pills').tagName).toMatch('SPAN');
    });
  });

  describe('Pills Component with overwrite class', () => {
    it('overwrite Pills class', () => {
      const { getByTestId } = render(<Pills className="PillsClass">{'Pills'}</Pills>);
      expect(getByTestId('DesignSystem-Pills')).toHaveClass('PillsClass');
    });
  });

  describe('Pills component with prop:appearance', () => {
    appearances.forEach((appearance) => {
      it(`should have the Badge--${appearance} class when appearance=${appearance} `, () => {
        const { getByTestId } = render(<Pills appearance={appearance}>{'Design'}</Pills>);
        expect(getByTestId('DesignSystem-Pills')).toHaveClass(`Badge--${appearance}`);
      });
    });
  });

  describe('Pills component with prop:subtle', () => {
    appearances.forEach((appearance) => {
      it(`should have the Badge--subtle-${appearance} class when appearance=${appearance} `, () => {
        const { getByTestId } = render(
          <Pills appearance={appearance} subtle={true}>
            {'Design'}
          </Pills>
        );
        expect(getByTestId('DesignSystem-Pills')).toHaveClass(`Badge--subtle-${appearance}`);
      });
    });
  });
});
