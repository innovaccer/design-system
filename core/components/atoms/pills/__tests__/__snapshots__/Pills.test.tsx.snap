// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pills component 
  appearance: "accent1", subtle: false
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--accent1"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "accent1", subtle: true
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--subtle-accent1"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "accent2", subtle: false
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--accent2"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "accent2", subtle: true
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--subtle-accent2"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "accent3", subtle: false
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--accent3"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "accent3", subtle: true
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--subtle-accent3"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "accent4", subtle: false
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--accent4"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "accent4", subtle: true
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--subtle-accent4"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "alert", subtle: false
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--alert"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "alert", subtle: true
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--subtle-alert"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "primary", subtle: false
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--primary"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "primary", subtle: true
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--subtle-primary"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "success", subtle: false
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--success"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "success", subtle: true
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--subtle-success"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "warning", subtle: false
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--warning"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;

exports[`Pills component 
  appearance: "warning", subtle: true
 1`] = `
<body>
  <div>
    <span
      class="Pills Badge--subtle-warning"
      data-test="DesignSystem-Pills"
    >
      Pills
    </span>
  </div>
</body>
`;
